awx/ui_next/src/screens/User/UserAccess/UserAccessList.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback, useEffect } from 'react';
+import { useParams, useLocation } from 'react-router-dom';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import { getQSConfig, parseQueryString } from '@util/qs';
+import { UsersAPI } from '@api';
+import useRequest from '@util/useRequest';
+import PaginatedDataList, {
+  ToolbarAddButton,
+} from '@components/PaginatedDataList';
+import DatalistToolbar from '@components/DataListToolbar';
+import UserAccessListItem from './UserAccessListItem';
+
+const QS_CONFIG = getQSConfig('roles', {
+  page: 1,
+  page_size: 20,
+  order_by: 'id',
+});
+// TODO Figure out how to best conduct a search of this list.
+// Since we only have a role ID in the top level of each role object
+// we can't really search using the normal search parameters.
+function UserAccessList({ i18n }) {
+  const { id } = useParams();
+  const { search } = useLocation();
+
+  const {
+    isLoading,
+    request: fetchRoles,
+    error,
+    result: { roleCount, roles, options },
+  } = useRequest(
+    useCallback(async () => {
+      const params = parseQueryString(QS_CONFIG, search);
+      const {
+        data: { results, count },
+      } = await UsersAPI.readRoles(id, params);
+      const {
+        data: { actions },
+      } = await UsersAPI.roleOptions(id);
+      return { roleCount: count, roles: results, options: actions };
+    }, [id, search]),
+    {
+      roles: [],
+      roleCount: 0,
+    }
+  );
+  useEffect(() => {
+    fetchRoles();
+  }, [fetchRoles]);
+  const canAdd =
+    options && Object.prototype.hasOwnProperty.call(options, 'POST');
+
+  const detailUrl = role => {
```

Comment:
This function needs it's own unit test that checks expected outputs for a set of inputs that trigger all 4 of the different branches.